### ISSUE TITLE
isEqual recursive loop

### ISSUE THREAD

**Issue by cobbs-totem** [OfficeDev/office-ui-fabric-react#4707]
### Bug Report

- __office-ui-fabric-react 5.92.0__: 

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? (Yes, No)

Requested priority: (Blocking, High, Normal, Low) 

Products/sites affected: (if applicable)

#### Describe the issue:

I'm not sure when or why things suddenly broke recently, but I noticed my <List> components throwing errors with the maximum stack exceeded during rendering.  I tracked it down to:

```
...
exports.isEqual: 62:1
compare 40:1
exports.isEqual 62:1
List.shouldComponentUpdate 208:1
```

I was able to pretty easily reproduce this by creating a brand new "carddemo" app using create-react-app.  The card demo renders fine, but if I replace the render() contents with something like:

```
<List 
  items={[ <span>hi</span> ]}
  onRenderCell={(item, index) => item}
/>
```

Eventually, the recursive isEqual gets to a couple of "FiberNode" elements and goes on forever until the exception is caught and thrown.

**Comment by dzearing** [comment]
@jordandrako is this related to your recent change?

**Comment by cliffkoh** [comment]
Fixed in `office-ui-fabric-react@5.92.1`
